src/Client.js

```diff
@@ -34,8 +33,7 @@ module.exports = class Client {
     ).then((r) => r.data.token_endpoint);
   }
 
-
-  generateClientAssertion(tokenEndpoint, jti) {
+  async generateClientAssetion(tokenEndpoint, jti) {
```

Comment:
Super minor typo in this function and its one usage later in the file, with this changeset. Should be:

```suggestion
  async generateClientAssertion(tokenEndpoint, jti) {
```